### PR TITLE
Chore: Reduce usage of v1 theme

### PR DESCRIPTION
**Pull request by kaydelaney**
**What this PR does / why we need it**:
Given that the old `GrafanaTheme` interface has been deprecated for a while, I thought it'd be a good idea to update any instances where it's still used to the current interface.

**Special notes for your reviewer**:

This is just one of a few PRs I have planned, but I didn't want any one PR to be too big so each will be around 40 files.

### CODE REVIEW

File: packages/grafana-data/src/themes/createV1Theme.ts [hunk 1]
```diff
@@ -250,3 +217,36 @@ const commonColorsPalette = {
   // New reds palette used by next-gen form elements
   red88: '#e02f44',
 };
+
+export const basicColors = {
+  ...commonColorsPalette,
+  black: '#000000',
+  white: '#ffffff',
+  dark1: '#141414',
+  dark2: '#161719',
```
Comment (joshhunt) [inline packages/grafana-data/src/themes/createV1Theme.ts:226]: hmmm, unsure about essentially removing these from the theme. 

I think they should have a home in Theme2 - probably identify where they're used and give them semantic names (in theme.components perhaps), or update where they're used to grab them from a more suitable place from Theme2.

Maybe for the time being the change to this file could/should be reverted, and consumers grab them from `theme2.v1.basicColors` or whatever, and tackle these separately - i think that would still be a marginal improvement.

File: public/app/plugins/panel/xychart/XYDimsEditor.tsx [hunk 2]
```diff
@@ -153,15 +152,15 @@ const getStyles = stylesFactory((theme: GrafanaTheme) => ({
   `,
 
   row: css`
-    padding: ${theme.spacing.xs} ${theme.spacing.sm};
-    border-radius: ${theme.border.radius.sm};
-    background: ${theme.colors.bg2};
-    min-height: ${theme.spacing.formInputHeight}px;
+    padding: ${theme.spacing(0.5)} ${theme.spacing(1)};
```
Comment (ashharrison90) [inline public/app/plugins/panel/xychart/XYDimsEditor.tsx:155]: minor nit: think you can do `${theme.spacing(0.5, 1)}`

File: public/app/plugins/panel/table/TablePanel.tsx [hunk 3]
```diff
@@ -133,7 +133,7 @@ export class TablePanel extends Component<Props> {
     }
 
     if (count > 1) {
-      const inputHeight = config.theme.spacing.formInputHeight;
+      const inputHeight = config.theme2.v1.spacing.formInputHeight;
```
Comment (ashharrison90) [inline public/app/plugins/panel/table/TablePanel.tsx:136]: minor nit: could we use `theme.components.height.md`?

File: packages/grafana-ui/src/components/DateTimePickers/RelativeTimeRangePicker/RelativeTimeRangePicker.tsx [hunk 4]
```diff
@@ -207,21 +207,21 @@ const getStyles = (fromError?: string, toError?: string) => (theme: GrafanaTheme
         cursor: pointer;
         padding-right: 0;
         padding-left: 0;
-        line-height: ${theme.v1.spacing.formInputHeight - 2}px;
+        line-height: ${theme.spacing.gridSize * 4 - 2}px;
```
Comment (ashharrison90) [inline packages/grafana-ui/src/components/DateTimePickers/RelativeTimeRangePicker/RelativeTimeRangePicker.tsx:210]: minor nit: could we use `${theme.components.height.md}`?

**Review by ashharrison90 — COMMENTED** [review]
this looks fantastic @kaydelaney 🙌 

added a couple of pretty minor nits.

re: removing the basic colors - i don't think they should be in there, at least as they are currently are. having named colors like `orange`/`yellow`/`purple` etc seems like a bad idea. so i don't mind removing these from the theme (as you've done) and then we can add things back in to theme2 gradually with better names?

i saw somewhere was using a red color... pretty sure that can be replaced with something like `theme.colors.error`... same for orange, there's something like `theme.colors.active`. i'd be surprised if there isn't already equivalents in most of theme2

File: packages/grafana-ui/src/themes/mixins.ts [hunk 5]
```diff
@@ -40,16 +40,16 @@ export function mediaUp(breakpoint: string) {
   return `only screen and (min-width: ${breakpoint})`;
 }
 
-export const focusCss = (theme: GrafanaTheme) => `
+export const focusCss = (theme: GrafanaTheme2) => `
```
Comment (jackw) [inline packages/grafana-ui/src/themes/mixins.ts:43]: Won't this break any plugin that uses this mixin? 🤔

File: public/app/plugins/panel/table/TablePanel.tsx (hunk 3 above)
Comment (kaydelaney) [reply public/app/plugins/panel/table/TablePanel.tsx:136]: yes we could! missed this one

File: packages/grafana-ui/src/themes/mixins.ts (hunk 5 above)
Comment (kaydelaney) [reply packages/grafana-ui/src/themes/mixins.ts:43]: You're right, missed this

**Review by ashharrison90 — APPROVED** [review]
lgtm 🙌

**Pull request merged**